doc/rst/platforms/infiniband.rst

```diff
@@ -230,29 +230,97 @@ an absolute value with something like
+the InfiniBand backend can do so by adding the following environment variable
+setting when building Chapel:
+
+.. code-block:: bash
+
+      export CHPL_GASNET_MORE_CFG_OPTIONS=--enable-pshm
+
+This will enable the PSHM support when co-locales are in-use, such that on-node
+communication between co-locales will occur in memory and not through the
+network.  It's worth noting that Chapel's integration of the PSHM feature
+currently requires an extra "progress" thread that defaults to running on the
+same cores as the Chapel tasks, and will compete with those tasks for cycles.
+For computation-bound applications the overhead incurred by this progress thread
+may outweigh the benefits of PSHM.  You can optionally disable PSHM at
+application run-time by setting ``export GASNET_SUPERNODE_MAXSIZE=1``, although
+this won't exactly match the behavior of building without PSHM support.
 
 Another alternative is to dedicate a core for the progress thread, preventing
 it from running on the same cores as the Chapel tasks. This is accomplished
 by setting ``CHPL_RT_COMM_GASNET_DEDICATED_PROGRESS_CORE=true``. Note that
 this means there will be one fewer core to run Chapel tasks, which may not be
 advantageous on machines with relatively few cores. Also note that this
 variable will dedicate a core whether or not PSHM is in-use, so you should
-only set this variable if you are using co-locales and you have not disabled
-PSHM using ``GASNET_SUPERNODE_MAXSIZE`` as described above.
+probably only set this variable if you are using co-locales and PSHM is enabled.
+
+----------------
+Progress Threads
+----------------
+
+In addition to the "busy" polling-based progress thread described above that
+is enabled when co-locales are combined with GASNet's (default disabled) PSHM
+support, the InfiniBand backend also optionally includes GASNet-level blocking
+progress threads used to help retire incoming and outgoing network communication
+operations (named the blocking "receive" and "send" progress threads, respectively). 
+These threads are "blocking" in that they are awakened on-demand when the network
+adapter reports there is communication work to be done, and otherwise generally
+remain parked in a kernel call where they do not consume any core cycles.
+By default the blocking receive progress thread is enabled and the blocking
+send progress thread is disabled.
+
+The blocking send progress thread may be enabled at application run-time by setting:
+
+.. code-block:: bash
+
+      export GASNET_SND_THREAD=1
+
+This enables a helper thread that has been shown to accelerate injection of
+communication in some cases, but in other cases may degrade communication
+throughput (notably on NUMA systems where the locale straddles the NUMA boundary).
+GASNet also provides additional environment variable settings that can
+optionally be used to control the detailed behavior of these threads, see
+GASNet documentation referenced in :ref:`ibv-conduit-docs`.
```

Comment:
One small nit, change to `...see the GASNet documentation referenced....`

Reply:
Resolved in 668f8e8054